pkg/liqoctl/common/cluster.go

```diff
@@ -0,0 +1,744 @@
+	if err != nil {
+		s.Fail(fmt.Sprintf("an error occurred while creating tenant namespace for remote cluster {%s}: %v", remoteClusterID.ClusterName, err))
+		return err
+	}
+	s.Success(fmt.Sprintf("tenant namespace {%s} created for remote cluster {%s}", ns.Name, remoteClusterID.ClusterName))
+	c.locTenantNamespace = ns.Name
+	return nil
+}
+
+// ExchangeNetworkCfg creates the local networkconfigs resource for the remote cluster, replicates it
+// into the remote cluster, waits for the remote cluster to populate the status of the resource and then
+// sets the remote status in the local networkconfigs resource.
+func (c *Cluster) ExchangeNetworkCfg(ctx context.Context, remoteClusterID *discoveryv1alpha1.ClusterIdentity) error {
+	// Enforce the network configuration in the local cluster.
+	s, _ := c.printer.Spinner.Start("creating network configuration in local cluster")
+	if err := c.enforceNetworkCfg(ctx, remoteClusterID, true); err != nil {
+		s.Fail(fmt.Sprintf("an error occurred while creating network configuration in local cluster: %v", err))
+		return err
+	}
+	s.Success(fmt.Sprintf("network configuration created in local cluster {%s}", c.clusterID.ClusterName))
+
+	// Enforce the network configuration in the local cluster.
+	s, _ = c.printer.Spinner.Start(fmt.Sprintf("creating network configuration in remote cluster {%s}", remoteClusterID.ClusterName))
+	if err := c.enforceNetworkCfg(ctx, remoteClusterID, false); err != nil {
+		s.Fail(fmt.Sprintf("an error occurred while creating network configuration in remote cluster: %v", err))
+		return err
+	}
+	s.Success(fmt.Sprintf("network configuration created in remote cluster {%s}", remoteClusterID.ClusterName))
+
+	// Wait for the network configuration to be processed by the remote cluster.
+	s, _ = c.printer.Spinner.Start(fmt.Sprintf("waiting network configuration to be processed by remote cluster {%s}", remoteClusterID.ClusterName))
+	netcfg, err := c.waitForNetCfg(ctx, remoteClusterID, 60*time.Second)
+	if err != nil {
+		s.Fail(err)
+		return err
+	}
+	s.UpdateText(fmt.Sprintf("reflecting network configuration status from cluster {%s}", remoteClusterID.ClusterName))
+	if err := c.reflectNetworkCfgStatus(ctx, remoteClusterID, &netcfg.Status); err != nil {
+		s.Fail(err)
+		return err
+	}
+	s.Success(fmt.Sprintf("network configuration status correctly reflected from cluster {%s}", remoteClusterID.ClusterName))
+	return nil
+}
+
+// enforceNetworkCfg enforces the presence of the networkconfigs resource for a given remote cluster.
+func (c *Cluster) enforceNetworkCfg(ctx context.Context, remoteClusterID *discoveryv1alpha1.ClusterIdentity, local bool) error {
+	// Get the network config.
+	netcfg, err := c.getNetworkCfg(ctx, remoteClusterID, local)
+	if client.IgnoreNotFound(err) != nil {
+		return err
+	}
+
+	if err != nil {
+		return c.createNetworkCfg(ctx, remoteClusterID, local)
+	}
+
+	return c.updateNetworkCfgSpec(ctx, netcfg, remoteClusterID, local)
```

Comment:
The mentioned function relies on the `namespace/name` key to retrieve the objects. Here we are using labels to get the resources in order to avoid to be bound to resource's name. The `controllerutils` functions are great when we already have the `object key` for the resource we are handling. Or am I missing something?

Reply:
ok, you are right 👍